packages/brick_offline_first/lib/src/offline_queue/offline_request_queue.dart

```diff
@@ -1,63 +1,55 @@
 import 'dart:async';
 import 'package:logging/logging.dart';
 import 'package:meta/meta.dart';
-import 'package:http/http.dart' as http;
-import 'offline_queue_http_client.dart';
 
 /// Repeatedly reattempts requests in an interval
-class OfflineRequestQueue {
+abstract class OfflineRequestQueue<_Client> {
   /// The client responsible for resending requests
-  final OfflineQueueHttpClient client;
+  final _Client client;
+
+  final String databaseName;
```

Comment:
`client` nor `databaseName` is used in this abstract class. 
```suggestion
abstract class OfflineRequestQueue {
```